dropwizard-configuration/src/main/java/io/dropwizard/configuration/JsonConfigurationFactory.java

```diff
@@ -0,0 +1,232 @@
+package io.dropwizard.configuration;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import com.fasterxml.jackson.databind.node.TreeTraversingParser;
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validator;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A factory class for loading JSON configuration files, binding them to configuration objects, and
+ * validating their constraints. Allows for overriding configuration parameters from system properties.
+ *
+ * @param <T> the type of the configuration objects to produce
+ */
+public class JsonConfigurationFactory<T> implements ConfigurationFactory<T> {
```

Comment:
> It feels wrong to have a json extend yaml
I also thought about that, should be other way around. The base factory approach looks even better to me.